lightly/api/bitmask.py

```diff
@@ -173,6 +186,34 @@ def intersection(self, other):
         """
         self.x = _intersection(self.x, other.x)
 
+    def difference(self, other):
+        """Calculates the difference of two bit masks.
+        Examples:
+            >>> mask1 = BitMask.from_bin('0b0111')
+            >>> mask2 = BitMask.from_bin('0b1100')
+            >>> mask1.difference(mask2)
+            >>> # mask1.binstring is '0b0011'
```

Comment:
This implementation is highly buggy. I quickly found two breaking examples.

Example 1
```
x = 7  (0b0111)
y = 12 (0b1100)
expected result: 0b0011
actual result: -0b0101 ( = bin(-5))
``` 

Example 2
```
x = 23 (0b10111)
y = 12 (0b01100)
expected result: 0b10011
actual result: 0b01011
```

I am pretty sure that the difference of the representing integers is not the way to get the difference of the two sets. In my opinion, you should simply mask all bits in `x` which are set in `y` so it would be something like:
```
return _union(x, y) - y
```

(the above code worked for my examples)